src/NoteI2c_Arduino.cpp

```diff
@@ -87,12 +87,19 @@ NoteI2c_Arduino::receive (
         } else if (response_length != request_length) {
             result = ERRSTR("serial-over-i2c: unexpected raw byte count {io}",i2cerr);
         } else {
-            // Update available with remaining bytes
-            *available_ = _i2cPort.read();
+            // Ensure available byte count is within expected range
+            static const size_t AVAILBLE_MAX = (NoteI2c::REQUEST_MAX_SIZE - NoteI2c::REQUEST_HEADER_SIZE);
+            uint32_t available = _i2cPort.read();
+            if (available > AVAILBLE_MAX) {
+                result = ERRSTR("serial-over-i2c: unexpected available byte count {io}",i2cerr);
```

Comment:
I think it would be better to say the available is greater than the max, rather than saying unexpected, which is more vague.